### DISCUSSION TITLE
How to load the flow into its initial viewport?

### DISCUSSION THREAD

**Discussion by wangjia184** [xyflow/xyflow#4651] [Q&A] [answered]
When SvelteFlow loads previously stored `nodes` and `edges`,   the nodes' relative postions are loaded correctly.
However, the viewport is not the same as before.  

How can I set viewport to the exact same status of previous state when `nodes` and `edges` are stored?
should I restore `x` `y` and `zoom` from `useViewport` ?

How can I set their initial values when SvelteFlow is loaded?

**Comment by bcakmakoglu** [comment, marked as answer]
`useSvelteFlow` provides a function called [`toObject`](https://svelteflow.dev/api-reference/hooks/use-svelte-flow#to-object), which you can use to save nodes, edges *and* the viewport.

You can then pass the saved `viewport` values to the [`initialViewport`](https://svelteflow.dev/api-reference/svelte-flow#initial-viewport) prop to restore it to the way it was when saved.
